src/objects/zcl_abapgit_object_intf.clas.abap

```diff
@@ -324,14 +329,33 @@ CLASS zcl_abapgit_object_intf IMPLEMENTATION.
                                            iv_clsname = ls_clskey-clsname ).
 
     " HERE: switch with feature flag for XML or JSON file format
-    io_xml->add( iv_name = 'VSEOINTERF'
-               ig_data = ls_intf-vseointerf ).
-    io_xml->add( iv_name =  'DESCRIPTIONS'
-                 ig_data = ls_intf-description ).
-    io_xml->add( iv_name = 'LINES'
-                 ig_data = ls_intf-docu-lines ).
-    io_xml->add( iv_name = 'I18N_LINES'
-                 ig_data = ls_intf-docu-i18n_lines ).
+    IF zcl_abapgit_persist_factory=>get_settings( )->read( )->get_experimental_features( ) = abap_true.
+      CREATE OBJECT lo_aff_mapper TYPE lcl_aff_type_mapping.
+      lo_aff_mapper->to_aff( EXPORTING iv_data = ls_intf
+                             IMPORTING es_data = ls_aff_intf ).
+      CREATE OBJECT lo_ajson.
+
+      TRY.
+
+          lv_json_xstring = lo_ajson->serialize( ls_aff_intf ).
+
+          zif_abapgit_object~mo_files->add_raw( iv_ext = 'json'
+                                                iv_data = lv_json_xstring ).
+
+        CATCH cx_root INTO lx_exception.
+          zcx_abapgit_exception=>raise_with_text( lx_exception ).
+      ENDTRY.
```

Comment:
put all this into the local class? a static method is okay from my POV

as things evolve, we might need to change this part, putting it to the local class reduces that impact